### PR TITLE
Cavs clk defs - deduplication and cleanup

### PR DESCRIPTION
**Pull request by mmaka1**
Cleanup of clock defs for cavs platforms, defs related to run-time data separated from platform capabilities to improve code readability. Common defs deduplicated for cAVS platforms.

### CODE REVIEW

File: src/platform/intel/cavs/include/cavs/clk.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	io_reg_update_bits(SHIM_BASE + SHIM_CLKCTL, SHIM_CLKCTL_HDCS, 0);
+#endif
+	io_reg_update_bits(SHIM_BASE + SHIM_CLKCTL,
+			   SHIM_CLKCTL_DPCS_MASK(cpu_get_id()),
+			   cpu_freq_enc);
+
+	return 0;
+}
+
+static inline int clock_platform_set_ssp_freq(uint32_t ssp_freqenc)
```
Comment (plbossart) [inline src/platform/intel/cavs/include/cavs/clk.h]: ssp_freq_enc?

File: src/platform/apollolake/include/platform/clk-map.h [hunk 2]
```diff
@@ -33,19 +33,26 @@
 
 #include <sof/clk.h>
 
+#define CLK_MAX_CPU_HZ		400000000
+#define CLK_DEFAULT_CPU_HZ	CLK_MAX_CPU_HZ
```
Comment (jajanusz) [inline src/platform/apollolake/include/platform/clk-map.h]: Can we get rid of `CLK_DEFAULT_CPU_HZ` at all? It's not used on cavs platforms.

**Comment by tlauda** [comment]
@mmaka1 ICL compilation is failing.

File: src/platform/apollolake/include/platform/clk-map.h (hunk 2 above)
Comment (mmaka1) [reply src/platform/apollolake/include/platform/clk-map.h]: Removed

File: src/platform/intel/cavs/include/cavs/clk.h (hunk 1 above)
Comment (mmaka1) [reply src/platform/intel/cavs/include/cavs/clk.h]: Fixed

**Comment by mmaka1** [comment]
> @mmaka1 ICL compilation is failing.

Right, there is still this old version of pm code running on those platforms and require clk-map.h (temporarily I believe).

**Pull request merged**